### DISCUSSION TITLE
What is the purpose of the SRGB_CORRECTION_FRAGMENT_SHADER?

### DISCUSSION THREAD

**Discussion by xujunjie12345** [alvr-org/ALVR#2509] [Q&A] [answered]
1. As far as I understand, textures in sRGB format are automatically converted to linear format when they are read into the shader for computation.
2. Why do other devices require manual conversion of the sRGB format, while Lynx does not?

**Comment by xujunjie12345** [comment]
I have figured out the first question. Whether it's OpenGL ES or D3D, when reading sRGB space textures, they are automatically converted to linear space values for computation. Therefore, in most cases, we do not need to manually perform color space conversion.

The decoded raw data is placed into an OES texture, which is a special type of texture that cannot be converted to a different color space during shader processing. Therefore, manual conversion is required in the SRGB_CORRECTION_FRAGMENT_SHADER.

**Reply by zmerp** [reply, marked as answer]
I think by spec OES samplers should do the necessary color conversions, but this should be implemented by hardware vendors (Qualcomm in this case) and rarely hardware vendors follow the specs.
